keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -36,6 +37,66 @@ class SqlDeliveryConfigRepository(
 
   private val mapper = configuredObjectMapper()
 
+  override fun deleteByApplication(application: String): Int {
+
+    val deliveryConfigUIDs = getUIDsByApplication(application)
+    val environmentUIDs = getEnvironmentUIDs(deliveryConfigUIDs)
+
+    deliveryConfigUIDs.forEach { uid ->
+      jooq.deleteFrom(DELIVERY_CONFIG)
+        .where(DELIVERY_CONFIG.UID.eq(uid))
+        .execute()
+    }
+
+    environmentUIDs.forEach { uid ->
+      jooq.deleteFrom(ENVIRONMENT)
+        .where(ENVIRONMENT.UID.eq(uid))
+        .execute()
+    }
+
+    deliveryConfigUIDs.forEach { uid ->
+      jooq.deleteFrom(DELIVERY_CONFIG_ARTIFACT)
+        .where(DELIVERY_CONFIG_ARTIFACT.DELIVERY_CONFIG_UID.eq(uid))
+        .execute()
+    }
+
+    deliveryConfigUIDs.forEach { uid ->
+      jooq.deleteFrom(DELIVERY_CONFIG_LAST_CHECKED)
+        .where(DELIVERY_CONFIG_LAST_CHECKED.DELIVERY_CONFIG_UID.eq(uid))
+        .execute()
+    }
+
+    environmentUIDs.forEach { uid ->
+      jooq.deleteFrom(ENVIRONMENT_ARTIFACT_VERSIONS)
+        .where(ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID.eq(uid))
+        .execute()
+    }
+
+    environmentUIDs.forEach { uid ->
```

Comment:
Minor nit, but I think this section would be a tad more readable if it were grouped into one each of `deliveryConfigUIDs.forEach{ }` and `environmentUIDs.forEach{ }`

Reply:
That was a great suggestion, thanks! and fixed :)